src/main/java/com/google/starfish/servlets/HandleNotesServlet.java

```diff
@@ -1,58 +1,129 @@
+
+    String title = request.getParameter("title");
     String school = request.getParameter("school").toLowerCase();
     String course = request.getParameter("course").toLowerCase();
     String[] miscLabels = request.getParameter("miscLabels").split(",");
 
-    // TODO: Put labels into database
-
-    // TODO: Put note into database
+    DataSource pool = (DataSource) request.getServletContext().getAttribute("my-pool");
 
-    response.setStatus(HttpServletResponse.SC_OK);
+    // Insert all labels before inserting note due to foreign key constraint
+    labelService.insertSchoolLabel(pool, school);
+    labelService.insertCourseLabel(pool, course);
+    for (String miscLabel : miscLabels) {
+      labelService.insertMiscLabel(pool, miscLabel.toLowerCase());
+    }
+    
+    try (Connection conn = pool.getConnection()) {
+      String stmt =
+          "INSERT INTO notes ( "
+              + "author_id,"
+              + "school,"
+              + "course,"
+              + "title,"
+              + "source_url,"
+              + "pdf_source,"
+              + "date_created,"
+              + "num_downloads ) "
+        + "VALUES ( "
+              + "?,"
+              + "?,"
+              + "?,"
+              + "?,"
+              + "?,"
+              + "?,"
+              + "?,"
+              + "? ); ";
+
+      try (PreparedStatement noteStmt = conn.prepareStatement(stmt, new String[] {"id"})) {
+        noteStmt.setString(1, userId);
+        noteStmt.setString(2, school);
+        noteStmt.setString(3, course);
+        noteStmt.setString(4, title);
+        if (sourceUrl == null) {
+          // If there was no sourceUrl, this is an uploaded pdf so
+          // set the sourceUrl to be the same as the pdfSource
+          noteStmt.setString(5, blobKey);
+        } else {
+          // If there is a sourceUrl on the request param, this is
+          // a google doc set set the source url based on the passed param
+          noteStmt.setString(5, sourceUrl);
+        }
+        // The blob key is nullable on the notes table so if this is a google doc
+        // this will be null
+        noteStmt.setString(6, blobKey);
+        noteStmt.setDate(7, new Date(Calendar.getInstance().getTimeInMillis()));
+        noteStmt.setLong(8, 0);
+        int rowsAffected = noteStmt.executeUpdate();
+        if (rowsAffected == 1) {
+          long noteId = 0;
+          ResultSet rs = noteStmt.getGeneratedKeys();
+          if (rs.next()) {
+            noteId = rs.getLong(1);
+            // Associate misc labels to the newly added note
+            for (String miscLabel : miscLabels) {
+              miscNoteLabelService.insertMiscNoteLabel(pool, noteId, miscLabel);
+            }
+          } 
+        }
+      }
+    } catch (SQLException ex) {
+      response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+      response.getWriter().println("INTERNAL SERVER ERROR: " + ex);
+    }    
+    response.sendRedirect("/");
```

Comment:
I don't think we need to the redirect anymore since the form will just close after it is submitted

Reply:
Ahh good point, I'll remove it then!